src/libstd/fs.rs

```diff
@@ -262,6 +262,73 @@ impl File {
         OpenOptions::new().write(true).create(true).truncate(true).open(path.as_ref())
     }
 
+    /// Read the entire contents of a file into a bytes vector.
+    ///
+    /// This is a convenience function for using [`File::open`] and [`read_to_end`]
+    /// with fewer imports and without an intermediate variable.
+    ///
+    /// [`File::open`]: struct.File.html#method.open
+    /// [`read_to_end`]: ../io/trait.Read.html#method.read_to_end
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if `path` does not already exist.
+    /// Other errors may also be returned according to [`OpenOptions::open`].
+    ///
+    /// [`OpenOptions::open`]: struct.OpenOptions.html#method.open
+    ///
+    /// It will also return an error if it encounters while reading an error
+    /// of a kind other than [`ErrorKind::Interrupted`].
+    ///
+    /// [`ErrorKind::Interrupted`]: ../../std/io/enum.ErrorKind.html#variant.Interrupted
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// #![feature(file_read_write_contents)]
+    ///
+    /// use std::fs::File;
+    ///
+    /// # fn foo() -> Result<(), Box<std::error::Error + 'static>> {
+    /// let foo = String::from_utf8(File::read_contents("foo.txt")?)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[unstable(feature = "file_read_write_contents", issue = /* FIXME */ "0")]
+    pub fn read_contents<P: AsRef<Path>>(path: P) -> io::Result<Vec<u8>> {
+        let mut bytes = Vec::new();
```

Comment:
This could look up the file size first and use `Vec::with_capacity`. It might mean as extra system call when reading a tiny file, but it is a [huge speedup][1] (up to 2× on my system) when reading large files.

[1]: https://github.com/TeXitoi/benchmarksgame-rs/issues/43

Reply:
Maybe `read_to_end` should be responsible for that?

Reply:
Specialization of `read_to_end` is a good idea. Filed #45851.

Note that callers of `read_to_end` can already avoid quadratic behavior by passing in a pre-allocated buffer.  That workaround isn't available with the functions in this PR.

Reply:
I’ve opened a PR with a prototype: https://github.com/rust-lang/rust/pull/45928